### PR TITLE
[CSL-2072] - Add voice search

### PR DESCRIPTION
**Pull request by mocca102**
API Documentation
https://docs.constructor.io/rest_api/search/natural_language_search/

Removed parameters because of no API support

- filters
- sort_order
- sort_by

Missing tests

- redirect_rule => if supported

### CODE REVIEW

File: src/types/search.d.ts [hunk 1]
```diff
@@ -40,6 +40,12 @@ declare class Search {
     userParameters?: UserParameters,
     networkParameters?: NetworkParameters
   ): Promise<SearchResponse>;
+
+  getVoiceSearchResults(
+    query: string,
+    parameters?: Omit<SearchParameters, 'filters' | 'sortBy' | 'sortOrder'>,
+    networkParameters?: NetworkParameters
```
Comment (Mudaafi) [inline src/types/search.d.ts:47]: I think we missed out userParameters here

File: src/modules/search.js [hunk 2]
```diff
@@ -255,6 +257,105 @@ class Search {
...
+          });
+        }
+
+        return json;
+      }
+
+      // Redirect rules
+      if (json.response && json.response.redirect) {
+        return json;
+      }
```
Comment (Mudaafi) [inline src/modules/search.js:354]: Haven't had a chance to test this but wanted to ask if a response had a redirect field, is it guaranteed not to have a results field? If it does, then this section becomes unreachable.

**Comment by Mudaafi** [comment]
Oh, could we also add `original_query?: string;` to `SearchRequestType`? Noticed you added it to the js-client repo but it was missing here.

File: src/types/search.d.ts (hunk 1 above)
Comment (mocca102) [reply src/types/search.d.ts:47]: Nice catch!

**Comment by mocca102** [comment]
> Oh, could we also add `original_query?: string;` to `SearchRequestType`? Noticed you added it to the js-client repo but it was missing here.

🦅

**Review by esezen — CHANGES_REQUESTED** [review]
Thing I mentioned on Slack

**Review by esezen — APPROVED** [review]
LGTM! @jjl014 would you like to take a final look or am I good to merge&deploy?

File: src/modules/search.js (hunk 2 above)
Comment (esezen) [reply src/modules/search.js:354]: @Mudaafi  Yes if there is a redirect, the `results` field will be omitted and the response object would only have `redirect`

**Comment by jjl014** [comment]
LGTM! Feel free to merge+deploy @esezen! 🚀

**Pull request merged**